composition-js/src/validate.ts

```diff
@@ -731,20 +1031,34 @@ class ValidationTraversal {
       subgraphNameToGraphEnumValue,
     );
   }
-  
+
   pushStack(state: ValidationState): { error?: GraphQLError } {
-    this.totalValidationSubgraphPaths += state.subgraphPathInfos.length;
+    if (state.subgraphPathInfos instanceof Array) {
```

Comment:
nit: Why don't we add a method to abstract the number of paths calculation (similar to `allSubgraphPathInfos`)? That will simplify `pushStack` and `popStack` a bit.